runtime/parachains/src/runtime_api_impl/vstaging.rs

```diff
@@ -16,12 +16,22 @@
 
 //! Put implementations of functions from staging APIs here.
 
-use crate::disputes;
-use primitives::v2::{CandidateHash, DisputeState, SessionIndex};
+use crate::{disputes, session_info, shared};
+use primitives::{
+	v2::{CandidateHash, DisputeState, SessionIndex},
+	vstaging::ExecutorParams,
+};
 use sp_std::prelude::*;
 
 /// Implementation for `get_session_disputes` function from the runtime API
 pub fn get_session_disputes<T: disputes::Config>(
 ) -> Vec<(SessionIndex, CandidateHash, DisputeState<T::BlockNumber>)> {
 	<disputes::Pallet<T>>::disputes()
 }
+
+/// Get session executor parameter set by parent hash
+pub fn session_ee_params_by_parent_hash<T: session_info::Config>(// parent_hash: T::Hash,
+) -> Option<ExecutorParams> {
+	let session_index = <shared::Pallet<T>>::session_index();
+	<session_info::Pallet<T>>::session_ee_params(session_index)
+}
```

Comment:
Will this always get the current session's params? What if we want to use the previous session's executor environment?

Also, should this be renamed i.e. remove the `_by_parent_hash` part, since we removed the `parent_hash` parameter?

Reply:
The last two weeks affected my understanding of things dramatically so please neglect my previous explanation of what's going on here :)

When a request to the runtime API is made, parent block hash is always passed as an implicit parameter. That hash specifies the state we're referring to, and the "state" concept is somewhat wider than I thought before. So when we're referring to some "state", we're asking runtime to behave as if it were in that state at the time of call, including the storage state. That is, if we're at block 10 and we're asking the runtime to execute something in the context of block 7, it will do it exactly as if it were at block 7. So the session index will also be correct in that case. Still looks like kind of magic to me but I've made @eskimor to swear that it works like that and he did.

That is something I carried out from my talks with @eskimor, @ordian and @sandreim. Let them correct me if I'm still missing something.

NB: It would be really great if we had those complex concepts explained in plain English somewhere (well, I do not insist on plain English, some kind of "engineering English" would do, but anyway it would be immensely useful for newcomers).

The part of your comment regarding renaming of things is totally legit but I'm thinking on global renaming. It is a total mess right now, I have `ExecutorParams` here and `ee_params` there and `EEPAR_` prefix in constants which do not add to readability, and also @vstakhov pointed out that `ee` thing is not very self-explaining for someone who's diving into the code for the first time. So I'm searching for sort of compromise to make it more readable, understandable and homogeneous and still avoid Java-like names as `EXECUTION_ENVIRONMENT_PARAMETER_NATIVE_STACK_LIMIT_IN_BYTES`. Any help is appreciated, right now I'm having something like `ExecEnvParams` and `exec_env_params` in my head but still not sure.

Reply:
Thanks for answering! `ExecEnvParams` seems clear to me FWIW.

Reply:
Ended up with `ExecutorParams` and `executor_params` everywhere. Rationale: although we're calling it "executor environment", "execution environment" or "executor" throughout discussions, the thing we're parametrizing it eventually called "Substrate Executor", so such a naming seems to me to be the most consistent.

Reply:
Makes sense.